tests/test_downscoped.py

```diff
@@ -383,3 +410,256 @@ def test_to_json(self):
                 ]
             }
         }
+
+
+class TestCredentials(object):
+    QUOTA_PROJECT_ID = "QUOTA_PROJECT_ID"
+    GRANT_TYPE = "urn:ietf:params:oauth:grant-type:token-exchange"
+    REQUESTED_TOKEN_TYPE = "urn:ietf:params:oauth:token-type:access_token"
+    TOKEN_EXCHANGE_ENDPOINT = "https://sts.googleapis.com/v1/token"
+    SUBJECT_TOKEN_TYPE = "urn:ietf:params:oauth:token-type:access_token"
+    SUCCESS_RESPONSE = {
+        "access_token": "ACCESS_TOKEN",
+        "issued_token_type": "urn:ietf:params:oauth:token-type:access_token",
+        "token_type": "Bearer",
+        "expires_in": 3600,
+    }
+    ERROR_RESPONSE = {
+        "error": "invalid_grant",
+        "error_description": "Subject token is invalid.",
+        "error_uri": "https://tools.ietf.org/html/rfc6749",
+    }
+    CREDENTIAL_ACCESS_BOUNDARY_JSON = {
+        "accessBoundary": {
+            "accessBoundaryRules": [
+                {
+                    "availablePermissions": AVAILABLE_PERMISSIONS,
+                    "availableResource": AVAILABLE_RESOURCE,
+                    "availabilityCondition": {
+                        "expression": EXPRESSION,
+                        "title": TITLE,
+                        "description": DESCRIPTION,
+                    },
+                }
+            ]
+        }
+    }
```

Comment:
Is there a particular reason to have these constants at class scope, rather than at module scope?

Reply:
I use module scope constants when I need them in multiple test classes (such as the ones above). In this case, they are used for a single test class. I can move them to module level but there is no consistent style in the repo.
Example: https://github.com/googleapis/google-auth-library-python/blob/d3944af36737dd7dc6a2a96932de5ae54f4e218a/tests/test_impersonated_credentials.py#L98

Reply:
I moved them to module scope.